### PR TITLE
Auth updates for login page + nginx conf + aaad

### PR DESCRIPTION
**Pull request by ankanm**
- [x] Added support for login page (with actas textbox).
- [x] Updated nginx configs
- [x] Updated aaad for auth support (I want to make more changes to this one but that's a different story.)

@vmahedia 
@amitbose327

### CODE REVIEW

File: ansible/roles/nginx/templates/rogeros_auth_proxy.conf.j2 [hunk 1]
```diff
@@ -9,6 +7,192 @@ upstream chronos {
...
+
+        ssl_certificate         {{ nginx_certs_dir }}/nginx.in.crt;
+        ssl_certificate_key     {{ nginx_certs_dir }}/nginx.in.key;
+        ssl_protocols           TLSv1 TLSv1.1 TLSv1.2;
+        ssl_ciphers             HIGH:!aNULL:!MD5;
+
+
+        #proxy_ssl_trusted_certificate /etc/ssl/certs/trusted_ca_cert.crt;
+
+        #proxy_ssl_verify        on;
```
Comment (amitbose327) [inline ansible/roles/nginx/templates/rogeros_auth_proxy.conf.j2]: We could ignore these commented directives for now as they depend on the "proxy_ssl_trusted_certificate" to be used.

File: aaad/aaad.py [hunk 2]
```diff
@@ -31,33 +40,62 @@ def do_GET(self):
...
+        # Read the seesion id cookie if available
+        cookie_str = self.headers.get('Cookie')
+        if cookie_str:
+            cookie_obj = SimpleCookie(cookie_str)
+            sid_morsel = cookie_obj.get(session_id_key, None)
+            if sid_morsel is not None:
+                sid = sid_morsel.value
+                if sid == '0':
+                    sid = None
+            else:
```
Comment (amitbose327) [inline aaad/aaad.py]: We don't need this and the next "else" statements

File: aaad/aaad.py [hunk 3]
```diff
@@ -31,33 +40,62 @@ def do_GET(self):
...
+
+        if sid: # If session_id_key cookie exists use that
+            decoded = self.decode_sessionid(sid)
+            if not decoded:
+                self.send_response(401)
+                self.send_header('Set-Cookie', '{}=0'.format(session_id_key)) # remove session id cookie
+                self.end_headers()
+                return
+            user = decoded['user']
+            password = decoded['password']
```
Comment (vmahedia) [inline aaad/aaad.py]: both `else` are redundant, we have initialized it to `None` anyway.

File: aaad/aaad.py (hunk 2 above)
Comment (vmahedia) [inline aaad/aaad.py]: both else are redundant, we have initialized it to None anyway.

File: ansible/roles/nginx/templates/rogeros_auth_proxy.conf.j2 [hunk 4]
```diff
@@ -154,11 +219,11 @@ server {
                 internal;
         }
 
-	location = /auth-proxy {
+        location = /auth-proxy {
                 proxy_pass http://localhost:{{ nginx_aaad_port }};
```
Comment (vmahedia) [inline ansible/roles/nginx/templates/rogeros_auth_proxy.conf.j2]: Format space.

**Pull request merged**